### ISSUE TITLE
Track Ninja's dyndeps feature

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/shake#357]
Ninja 1.7 is adding dyndeps (will be in the Ninja manual once it's done), along with examples of extracting a tarball and compiling Fortran. I should make documentation on the website of how to do that, and also support dyndeps in the Ninja build.

**Comment by Mathnerd314** [comment]
Ninja documentation link: https://github.com/bradking/ninja/blob/features-for-fortran/doc/manual.asciidoc#dynamic-dependencies

It's not clear if it will actually be in 1.7.

**Comment by Mathnerd314** [comment]
The tarball example is like the zip file in https://github.com/ndmitchell/shake/issues/421#issuecomment-184412157.

I'm not sure about the Fortran example; it seems like it would ``fscan`` all the files on every build, while in fact you only need to ``fscan`` the changed files.

**Comment by ndmitchell** [comment]
Yep, if it goes in, I'd aim to support it. I find the dyndeps thing a little confusing at first glance - it seems like a weird primitive to have added (I assume its a restricted version of monadic dependencies in some way).

**Comment by Mathnerd314** [comment]
Dynamic implicit inputs can be replicated by a function that reads the dyndep file and calls ``need``.

Dynamic implicit outputs are more interesting; they're a ``provide``-like operation, with the semantics that modifying a ``provide``-d file will re-run the provider.

**Comment by Ericson2314** [comment]
https://github.com/ninja-build/ninja/pull/1521 wow it actually landed! The ranks of monadic build systems have grown.
